src/Cli/dotnet/Commands/Test/MTP/MSBuildUtility.cs

```diff
@@ -18,28 +20,55 @@ internal static class MSBuildUtility
 
         if (!isBuiltOrRestored)
         {
             return (Array.Empty<ParallelizableTestModuleGroupWithSequentialInnerModules>(), isBuiltOrRestored);
         }
 
-        string rootDirectory = solutionFilePath.HasExtension(".slnf") ?
-                Path.GetDirectoryName(solutionModel.Description)! :
-                SolutionAndProjectUtility.GetRootDirectory(solutionFilePath);
+        var msbuildArgs = MSBuildArgs.AnalyzeMSBuildArguments(buildOptions.MSBuildArgs, CommonOptions.PropertiesOption, CommonOptions.RestorePropertiesOption, CommonOptions.MSBuildTargetOption(), CommonOptions.VerbosityOption());
+        var solutionFile = SolutionFile.Parse(solutionFilePath);
+        var globalProperties = CommonRunHelpers.GetGlobalPropertiesFromArgs(msbuildArgs);
 
-        FacadeLogger? logger = LoggerUtility.DetermineBinlogger([.. buildOptions.MSBuildArgs], dotnetTestVerb);
+        globalProperties.TryGetValue("Configuration", out var activeSolutionConfiguration);
+        globalProperties.TryGetValue("Platform", out var activeSolutionPlatform);
 
-        var msbuildArgs = MSBuildArgs.AnalyzeMSBuildArguments(buildOptions.MSBuildArgs, CommonOptions.PropertiesOption, CommonOptions.RestorePropertiesOption, CommonOptions.MSBuildTargetOption(), CommonOptions.VerbosityOption());
+        if (string.IsNullOrEmpty(activeSolutionConfiguration))
+        {
+            activeSolutionConfiguration = solutionFile.GetDefaultConfigurationName();
+        }
+
+        if (string.IsNullOrEmpty(activeSolutionPlatform))
+        {
+            activeSolutionPlatform = solutionFile.GetDefaultPlatformName();
+        }
+
+        var solutionConfiguration = solutionFile.SolutionConfigurations.FirstOrDefault(c => c.ConfigurationName == activeSolutionConfiguration && c.PlatformName == activeSolutionPlatform)
+            ?? throw new InvalidOperationException($"The solution configuration '{activeSolutionConfiguration}|{activeSolutionPlatform}' is invalid.");
+
+        // TODO: What to do if the given key doesn't exist in the ProjectConfigurations?
+        // MSBuild seems to be special casing web projects specifically!!
+        // https://github.com/dotnet/msbuild/blob/243fb764b25affe8cc5f233001ead3b5742a297e/src/Build/Construction/Solution/SolutionProjectGenerator.cs#L659-L672
+        // It doesn't make sense to have to duplicate the MSBuild logic here and having to maintain them in sync.
+        // https://github.com/dotnet/msbuild/issues/12711 tracks having a better public API.
+        var projectPaths = solutionFile.ProjectsInOrder
+            .Where(p => ProjectShouldBuild(solutionFile, p.RelativePath) && p.ProjectConfigurations.ContainsKey(solutionConfiguration.FullName))
+            .Select(p => (p.ProjectConfigurations[solutionConfiguration.FullName], p.AbsolutePath))
+            .Where(p => p.Item1.IncludeInBuild)
+            .Select(p => (p.AbsolutePath, p.Item1.ConfigurationName, p.Item1.PlatformName));
```

Comment:
@rainersigwald @baronfel Can `ConfigurationName` or `PlatformName` here be null or empty?